### PR TITLE
ECER-1834 - BE: Create endpoint to get application status

### PR DESCRIPTION
**Pull request by qsl-deepak**
---
name: ECER-1834 - BE: Create endpoint to get application status
about: BE: Create endpoint to get application status
---

## Title
ECER-1834 : BE- Create endpoint to get application status

## Description

- BE: Added endpoint and all layer mappings to get status of the submitted application
- BE: Added automated tests

## Related Jira Issue(s)

- ECER-1834
- ECER-1259

## Checklist
- [x] I have tested these changes locally.
- [x] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

## Additional Comments (optional)
Any additional context or information that might be helpful for the reviewer.

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using Microsoft.AspNetCore.Mvc;
 using System.ComponentModel;
 using System.ComponentModel.DataAnnotations;
+using System.Net.NetworkInformation;
```
Comment (ytqsl) [inline src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs:11]: ?

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs [hunk 2]
```diff
@@ -79,6 +80,31 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
         .RequireAuthorization()
         .WithParameterValidation();
 
+
+    endpointRouteBuilder.MapGet("/api/applications/{id}/status", async Task<Results<Ok<SubmittedApplicationStatus>, BadRequest<ProblemDetails>, NotFound<ProblemDetails>>> (string id, HttpContext ctx, IMediator messageBus, IMapper mapper, CancellationToken ct) =>
+    {
+      var userId = ctx.User.GetUserContext()?.UserId;
+      bool IdIsNotGuid = !Guid.TryParse(id, out _); if (IdIsNotGuid)
```
Comment (ytqsl) [inline src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs:87]: Not a syntax we generally use in C#. Make sure your auto formatting is enabled and that you've installed the right extensions

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs [hunk 3]
```diff
@@ -233,3 +261,27 @@ public record CharacterReference([Required] string? FirstName, [Required] string
 {
   public string? Id { get; set; }
 }
+
+public record SubmittedApplicationStatus(string Id, DateTime SubmittedOn, ApplicationStatus Status)
```
Comment (ytqsl) [inline src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs:265]: Why is this endpoiint and DTO back?

**Review by ytqsl — CHANGES_REQUESTED** [review]

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs (hunk 1 above)
Comment (qsl-deepak) [reply src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs]: Removed unused usings

**Review by ytqsl — APPROVED** [review]

**Pull request merged**